packages/host/app/services/card-service.ts

```diff
@@ -452,8 +452,18 @@ export default class CardService extends Service {
     return card[api.realmInfo];
   }
 
-  async getRealmURL(card: CardDef) {
+  async getRealmURL(card: CardDef, opts?: { isWritableRealm: boolean }) {
     let api = await this.getAPI();
+    if (opts?.isWritableRealm) {
+      let url = card?.[api.realmURL];
+      if (url && this.realm.canWrite(url.href)) {
+        return url;
+      }
+      if (!this.realm.defaultWritableRealm) {
+        throw new Error('Could not find a writable realm');
+      }
+      return new URL(this.realm.defaultWritableRealm.path);
+    }
     // in the case where we get no realm URL from the card, we are dealing with
     // a new card instance that does not have a realm URL yet.
     return card[api.realmURL] ?? new URL(this.realm.defaultReadableRealm.path);
```

Comment:
I see that this PR is extending the existing fallback behavior. But this fallback behavior doesn't belong here.

`getRealmURL` should only return a card's actual realm url. When there isn't one, it should return undefined, and it should be up to the caller to pick a default instead.